### PR TITLE
Ch setup continuous integration #24

### PR DESCRIPTION
**Pull request by ALVINdimpos**
#### What does this PR do?

This PR configures GitHub Actions as a continuous integration service to build and test the app, and adds the GitHub Actions badge to the repo.

#### Description of Task to be completed

- Configure GitHub Actions to run tests and build steps for the app.
- Add the GitHub Actions badge to the repo.

#### How should this be manually tested?

1. Clone the repo .
2. cd in project.
3. npm install.
4. then run npm run test
#### Any background context you want to provide?

We chose GitHub Actions as our continuous integration service because it integrates well with our GitHub repository and is easy to set up.

#### What are the relevant pivotal tracker stories?

CardId:#24

[Node.js CI](https://github.com/atlp-rwanda/ecommerce-app-predators-bn/actions/workflows/node.js.yml)

https://trello.com/c/a879sh8b/24-setup-continuous-integration-quality-assurance-pipeline
Deployed Link:https://ecommerce-app-predators-bn.onrender.com/
#### Screenshots (if appropriate)

N/A

### CODE REVIEW

**Review by faid-terence — COMMENTED** [review]
The setup is good but i have a question 
Why do you have only one folder ?

**Comment by ALVINdimpos** [comment]
> The setup is good but i have a question Why do you have only one folder ?
i do not need another file  to setup github action workflow

File: .github/workflows/node.js.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+# For more information see: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-nodejs
+
+name: E-commerce App
+version: 1.0.0
+description: A simple Node.js application for E-commerce App
+author: Team predetor
+
+on:
+  push:
+    branches: [ "*" ]
```
Comment (chanceDira) [inline .github/workflows/node.js.yml:4]: I believe having at least develop branch here is better

File: .github/workflows/node.js.yml [hunk 2]
```diff
@@ -0,0 +1,32 @@
+# This workflow will do a clean installation of node dependencies, cache/restore them, build the source code and run tests across different versions of node
+# For more information see: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-nodejs
+
+name: E-commerce App
+version: 1.0.0
+description: A simple Node.js application for E-commerce App
```
Comment (chanceDira) [inline .github/workflows/node.js.yml]: Remove the "simple" word

File: .github/workflows/node.js.yml [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+author: Team predetor
```
Comment (chanceDira) [inline .github/workflows/node.js.yml]: "Team Predators"

File: .github/workflows/node.js.yml [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    strategy:
+      matrix:
+        node-version: [14.x, 16.x, 18.x]
+        # See supported Node.js release schedule at https://nodejs.org/en/about/releases/
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+      - name: Install Dependencies
+        run: npm ci
```
Comment (chanceDira) [inline .github/workflows/node.js.yml]: installing dependencies you can use "npm i" instead is "npm ci"

**Review by chanceDira — CHANGES_REQUESTED** [review]
Good progress, you can create a sample test to allow this build job to work well for now

File: .github/workflows/node.js.yml (hunk 4 above)
Comment (ALVINdimpos) [reply .github/workflows/node.js.yml]: done

File: .github/workflows/node.js.yml (hunk 3 above)
Comment (ALVINdimpos) [reply .github/workflows/node.js.yml]: done

File: .github/workflows/node.js.yml (hunk 2 above)
Comment (ALVINdimpos) [reply .github/workflows/node.js.yml]: done

File: .github/workflows/node.js.yml (hunk 1 above)
Comment (ALVINdimpos) [reply .github/workflows/node.js.yml:4]: done

**Comment by ALVINdimpos** [comment]
> Good progress, you can create a sample test to allow this build job to work well for now

thanks! now solved

File: .github/workflows/node.js.yml
Comment (yobuir) [inline .github/workflows/node.js.yml:1]: nice setup 
I think there is remaining part of running the test of the app

**Review by yobuir — COMMENTED** [review]
nice setup
I think there is remaining part of running the test of the app

**Review by Muhet — COMMENTED** [review]
are we going tu test build only? what about run test and deployment jobs?

**Review by nili-alu — APPROVED** [review]
This is great job, just apply the feedback you have above

**Review by bryan936M — APPROVED** [review]
Congratulation on your successful completion of your task

**Review by yobuir — APPROVED** [review]

**Review by faid-terence — APPROVED** [review]

**Review by Muhet — APPROVED** [review]

**Comment by mukunzidd** [comment]
@ALVINdimpos please add the card link under: "What are the relevant pivotal tracker stories?" This should never be empty.

**Review by mukunzidd — CHANGES_REQUESTED** [review]
Alvin's PR
- @ALVINdimpos please work on your PR description. I'm sure this is not how one tests your chore locally.
- Leave instruction in the readme.md about how one is supposed to use the linting utility

**Comment by ALVINdimpos** [comment]
> Alvin's PR
> 
>     * @ALVINdimpos please work on your PR description. I'm sure this is not how one tests your chore locally.
> 
>     * Leave instruction in the readme.md about how one is supposed to use the linting utility

ok let me work on it.

**Comment by ALVINdimpos** [comment]
> @ALVINdimpos please add the card link under: "What are the relevant pivotal tracker stories?" This should never be empty.

done! now you can check

File: package.json [hunk 5]
```diff
@@ -15,8 +15,9 @@
     "unseed": "npx sequelize-cli db:seed:undo",
     "down": "npx sequelize-cli db:migrate:undo",
     "test": "cross-env NODE_ENV=test jest",
-    "start": "cross-env NODE_ENV=production node server.js",
-    "dev": "cross-env NODE_ENV=development nodemon --experimental-modules --es-module-specifier-resolution=node server.js",
+    "start": "NODE_ENV=development nodemon --exec babel-node src/server.js",
```
Comment (chanceDira) [inline package.json]: there is no cross-env on the start script

**Review by chanceDira — CHANGES_REQUESTED** [review]
Nice work, let's add deployment also to have this task finished.

File: package.json (hunk 5 above)
Comment (ALVINdimpos) [reply package.json]: done!

**Comment by ALVINdimpos** [comment]
> 



> Nice work, let's add deployment also to have this task finished.

done!

**Review by chanceDira — APPROVED** [review]
Good job @ALVINdimpos

File: .github/workflows/node.js.yml [hunk 6]
```diff
@@ -0,0 +1,40 @@
+name: My Workflow
```
Comment (mukunzidd) [inline .github/workflows/node.js.yml]: Please change the workflow name to CI.

**Review by mukunzidd — CHANGES_REQUESTED** [review]
Fix the workflow name. You can call it CI. If they are separate you can call on build and another one test. But everything else looks good.

File: .github/workflows/node.js.yml (hunk 6 above)
Comment (ALVINdimpos) [reply .github/workflows/node.js.yml]: done!

**Review by mukunzidd — APPROVED** [review]

**Pull request merged**